### ISSUE TITLE
Error in Microsoft Teams App on Outlook Mobile: Instantiating TeamsUserCredential Throws InternalError

### ISSUE THREAD

**Issue by akanjimm** [OfficeDev/TeamsFx#10987]
**Describe the bug**
Our Microsoft Teams app works properly on all other hosts but throws an error on Outlook mobile. 

The error occurs when instantiating the TeamsUserCrendential class and attempting to get the logged-in user's basic information (only on Outlook mobile).

**To Reproduce**
1. Scaffold a new sso enabled tab app (Teams, Outlook, Microsoft 365)
2. Instantiate the TeamsUserCredential class and call the getUserInfo() or getToken("") method

```
// Example code using TeamsFx SDK version 2.3.0
const { loading, data, reload } = useData(async () => {        
	try {            
		const authConfig = {                
			clientId: process.env.REACT_APP_CLIENT_ID,                
			initiateLoginEndpoint: process.env.REACT_APP_START_LOGIN_PAGE_URL,            
		};            
		const credential = new TeamsUserCredential(authConfig);            
		const userData = await credential.getUserInfo();        
	} catch (err) {            
		// error handling        
}});
```
3. Check out the app on Outlook mobile

**Expected behavior**
For the app to be accessible across all hosts (desktop, web and mobile) and to get a logged-in user's basic data, without throwing error on outlook mobile

**Screenshots**
Check out screenshot for error message on Outlook mobile and inability to get data from the Azure Functions backend
![MicrosoftTeams-image (2)](https://github.com/OfficeDev/TeamsFx/assets/81008428/7ea33ee1-1e04-4ec3-a3b7-c1bec6a07430)

Test app on Microsoft 365 mobile
![MicrosoftTeams-image (3)](https://github.com/OfficeDev/TeamsFx/assets/81008428/4072220a-da41-46be-8860-8d3f9d10ee79)

**Comment by KennethBWSong** [comment]
@akanjimm Thank you for reporting this issue. Would you please tell us which OS are you using (Android/iOS) and the version of Outlook app?

**Comment by akanjimm** [comment]
@KennethBWSong I'm using iOS and the version of the Outlook app is 4.2408.1

**Comment by KennethBWSong** [comment]
@akanjimm Thank you for your info. We are investigating this issue and will reach out to you as soon as we have any updates.

**Comment by petr-istratov** [comment]
Any updates in this? We're having same issue

**Comment by KennethBWSong** [comment]
Hi @petr-istratov We have upstream dependency on Outlook team for this issue and will update here if any progress.

**Comment by KennethBWSong** [comment]
Hi @akanjimm  @petr-istratov Sorry for late reply. After debug with Outlook team, would you please try to add following lines in `./aad.manifest.json` and see whether this can fix your issue?
```
{
  ...
  "preAuthorizedApplications": [
    ...
    {
      "appId": "27922004-5251-4030-b22d-91ecd9a37ea4",
      "permissionIds": [
        "${{AAD_APP_ACCESS_AS_USER_PERMISSION_ID}}"
      ]
    }
  ]
}
```

**Comment by petr-istratov** [comment]
@KennethBWSong Yes, thanks for the reply. This actually helps or it can be changed in App registration in Azure.
Apart from this change you need to reconfigure environment variables in the backend. Update `ALLOWED_APP_IDS` and `WEBSITE_AUTH_AAD_ACL`

**Comment by KennethBWSong** [comment]
@petr-istratov  Thanks for your info! We will also fix this issue in the TTK in the later release and update here when the fix is ready.

**Comment by mr-github-user** [comment]
> Hi @akanjimm @petr-istratov Sorry for late reply. After debug with Outlook team, would you please try to add following lines in `./aad.manifest.json` and see whether this can fix your issue?
> 
> ```
> {
>   ...
>   "preAuthorizedApplications": [
>     ...
>     {
>       "appId": "27922004-5251-4030-b22d-91ecd9a37ea4",
>       "permissionIds": [
>         "${{AAD_APP_ACCESS_AS_USER_PERMISSION_ID}}"
>       ]
>     }
>   ]
> }
> ```

I was having the same issue and this fixed it for me.
